File: advertiser.py
from base_model import *
class Advertiser(BaseAdvertising):
	'This calss is meant to represent an advertiser'
	totalClicks = 0
	def __init__(self, ID, NAME):
		self._id = ID
		self.__name = NAME
		self._clicks = self._views = 0
	def getName(self):
		return self.__name
	def setName(self, NAME):
		self.__name = NAME
	def getTotalClicks():
		return Advertiser.totalClicks
	def incClicks(self):
		self._clicks += 1
		Advertiser.totalClicks += 1
	def incViews(self):
		self._views += 1
	def describeMe(self):
		return Advertiser.__doc__
	def help():
		return """Here's a short description for each field of this class :
		id: The id of this advertiser.
		name: The name of the advertiser.
		totalClicks: The total number of clicks that were made.
		clicks: The number of clicks that were made on this advertiser's ads.
		views: The number of times one of this advertiser's ads was viewed."""

File: Cpp/advertiser.cpp
#include "base_model.cpp"
using namespace std;
class Advertiser: public BaseAdvertising {
private:
    string name;
public:
    static int totalClicks;
    Advertiser(int _id = 0, string _name = "") {
        id = _id; name = _name;
    }
    string getName() {
        return (name);
    }
    void setName(string _name) {
        name = _name;
    }
    static int getTotalClicks() {
        return (totalClicks);
    }
    void incClicks() {
        clicks ++;
        totalClicks ++;
    }
    void incView() {
        views ++;
    }
    string describeMe() {
        return ("This class is meant to represent an advertiser.");
    }
   static string help() {
        return ("Here's a short description for each field of this class :\n\
                id: The id of this advertiser.\n\
                name: The name of the advertiser.\n\
                totalClicks: The total number of clicks that were made.\n\
                clicks: The number of clicks that were made on this advertiser's ads.\n\
                views: The number of times one of this advertiser's ads was viewed.");
    }
};
